crypto/dsa/dsa_key.c

```diff
@@ -82,14 +84,74 @@ static int dsa_builtin_keygen(DSA *dsa)
 
     dsa->priv_key = priv_key;
     dsa->pub_key = pub_key;
-    dsa->dirty_cnt++;
+
+#ifdef FIPS_MODE
+    pairwise_test = 1;
+#endif /* FIPS_MODE */
+
     ok = 1;
+    if (pairwise_test) {
+        OSSL_CALLBACK *cb = NULL;
+        void *cbarg = NULL;
+
+        OSSL_SELF_TEST_get_callback(dsa->libctx, &cb, &cbarg);
+        ok = dsa_keygen_pairwise_test(dsa, cb, cbarg);
+        if (!ok) {
+            BN_free(dsa->pub_key);
+            BN_clear_free(dsa->priv_key);
+            BN_CTX_free(ctx);
+            return ok;
+        }
+    }
+    dsa->dirty_cnt++;
 
  err:
     if (pub_key != dsa->pub_key)
         BN_free(pub_key);
     if (priv_key != dsa->priv_key)
         BN_free(priv_key);
     BN_CTX_free(ctx);
+
     return ok;
 }
+
+static int dsa_builtin_keygen(DSA *dsa)
+{
+    return dsa_keygen(dsa, 0);
+}
```

Comment:
I didn't notice this before.  So you're renaming an *static* function, then creating a wrapper that's also a *static* function...

Why not just change the existing static function and change all calls from `dsa_builtin_keygen(dsa)` to `dsa_builtin_keygen(dsa, 0)`?

Reply:
I do understand the corresponding `ec_key_simple_generate_key` change, since the calls are so much more widely spread.  But when it's limited to the same file?  Less so...

Reply:
Its also just one call :) Changed